### PR TITLE
[BAHIR-247] Provide connection validation/idle testing for Flink-Redis Connector

### PR DESCRIPTION
**Pull request by YikSanChan**
The PR address @eskabetxe 's comment https://github.com/apache/bahir-flink/pull/104/files#r556403547

@eskabetxe Let me know if you have any other concern. If not, I will add comments to the code.

### CODE REVIEW

**Comment by eskabetxe** [comment]
@YikSanChan its the same commit? if yes could we maintain the original committer?

**Comment by YikSanChan** [comment]
@eskabetxe Hey it is very similar to the previous commit but address your comments. I tried to add an additional commit to the previous PR but I don't have write access to the user's repo.

**Comment by YikSanChan** [comment]
@eskabetxe Hi Joao, could you please suggest next steps? 🙏

**Comment by YikSanChan** [comment]
@eskabetxe Can you please walk me through how am I able to continue with the PR https://github.com/apache/bahir-flink/pull/104?

> If you left its branch

What does "left its branch" mean? This repo only has one branch `master` (https://github.com/apache/bahir-flink/branches), and lmagic233's branch is not in the current repo. I am not able to fetch his commits

```
> git remote add lmagic233 git@github.com:lmagic233/bahir-flink.git

> git remote -v
lmagic233	git@github.com:lmagic233/bahir-flink.git (fetch)
lmagic233	git@github.com:lmagic233/bahir-flink.git (push)
origin	https://github.com/YikSanChan/bahir-flink.git (fetch)
origin	https://github.com/YikSanChan/bahir-flink.git (push)
upstream	https://github.com/apache/bahir-flink.git (fetch)
upstream	https://github.com/apache/bahir-flink.git (push)

> git fetch lmagic233
The authenticity of host 'github.com (52.74.223.119)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added 'github.com,52.74.223.119' (RSA) to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by eskabetxe** [comment]
@YikSanChan I fix the original committer on this PR
could you check that I dont delete anything wrong or something get merged bad

after that I will check the PR

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/config/FlinkJedisClusterConfig.java
Comment (YikSanChan) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/config/FlinkJedisClusterConfig.java:246]: remove `'\''`

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/config/FlinkJedisPoolConfig.java
Comment (YikSanChan) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/config/FlinkJedisPoolConfig.java:258]: ditto

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/config/FlinkJedisSentinelConfig.java
Comment (YikSanChan) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/config/FlinkJedisSentinelConfig.java:286]: ditto

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/config/FlinkJedisSentinelConfig.java
Comment (YikSanChan) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/config/FlinkJedisSentinelConfig.java:294]: ditto

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisCommandsContainerBuilder.java
Comment (YikSanChan) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisCommandsContainerBuilder.java:74]: This is no longer needed since they are covered by `getGenericObjectPoolConfig`

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisCommandsContainerBuilder.java
Comment (YikSanChan) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisCommandsContainerBuilder.java:100]: ditto

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisCommandsContainerBuilder.java
Comment (YikSanChan) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisCommandsContainerBuilder.java:129]: ditto

**Review by YikSanChan — COMMENTED** [review]
@eskabetxe Commented, I will make a commit shortly

**Comment by eskabetxe** [comment]
@YikSanChan could you address those changes?

**Comment by YikSanChan** [comment]
@eskabetxe Work in progress

**Comment by YikSanChan** [comment]
@eskabetxe Fixed.

**Review by eskabetxe — CHANGES_REQUESTED** [review]
I would like to see some test with the new configuration, is this possible?

**Comment by YikSanChan** [comment]
@eskabetxe Regarding tests, I am thinking of adding a unit test for `RedisCommandsContainerBuilder`, and mainly test the `getGenericObjectPoolConfig` method. What do you think?

**Comment by YikSanChan** [comment]
@eskabetxe Hi Joao could you please see my questions above? Thanks!

**Comment by eskabetxe** [comment]
Hi @YikSanChan , sorry for the delay..
Any test added yes always welcome

**Comment by YikSanChan** [comment]
@eskabetxe Added tests in flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/common/container/RedisCommandsContainerBuilderTest.java, let me know if you have any other concern, thanks

**Comment by YikSanChan** [comment]
@eskabetxe BTW I wonder what is the release schedule? Can we release a new version of flink-connector-redis after merging this in?

**Comment by YikSanChan** [comment]
@eskabetxe Could you please help with another look? Thank you so much!

**Comment by eskabetxe** [comment]
@YikSanChan  thanks for your contribution
merged

**Pull request merged**
